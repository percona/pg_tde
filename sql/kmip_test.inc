CREATE EXTENSION pg_tde;

SELECT pg_tde_add_key_provider_kmip('kmip-prov','127.0.0.1', 5696, '/tmp/server_certificate.pem', '/tmp/client_key_jane_doe.pem');
SELECT pg_tde_set_principal_key('kmip-principal-key','kmip-prov');

CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING :tde_am;

INSERT INTO test_enc (k) VALUES (1);
INSERT INTO test_enc (k) VALUES (2);
INSERT INTO test_enc (k) VALUES (3);

SELECT * from test_enc;

DROP TABLE test_enc;

DROP EXTENSION pg_tde;
