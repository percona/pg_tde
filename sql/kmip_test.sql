\set tde_am tde_heap
\i sql/kmip_test.inc