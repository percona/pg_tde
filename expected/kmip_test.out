\set tde_am tde_heap
\i sql/kmip_test.inc
CREATE EXTENSION pg_tde;
SELECT pg_tde_add_key_provider_kmip('kmip-prov','127.0.0.1', 5696, '/tmp/server_certificate.pem', '/tmp/client_key_jane_doe.pem');
 pg_tde_add_key_provider_kmip 
------------------------------
                            1
(1 row)

SELECT pg_tde_set_principal_key('kmip-principal-key','kmip-prov');
 pg_tde_set_principal_key 
--------------------------
 t
(1 row)

CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING :tde_am;
INSERT INTO test_enc (k) VALUES (1);
INSERT INTO test_enc (k) VALUES (2);
INSERT INTO test_enc (k) VALUES (3);
SELECT * from test_enc;
 id | k 
----+---
  1 | 1
  2 | 2
  3 | 3
(3 rows)

DROP TABLE test_enc;
DROP EXTENSION pg_tde;
